Max/src/main/java/springbootApp/java/models/InstrumentoFinanciero.java

```diff
@@ -0,0 +1,64 @@
+package springbootApp.java.models;
+
+
+public abstract class InstrumentoFinanciero {
+
+    private String nombre;
+    private double precio;
+    private Tipo tipo;
+
+    public InstrumentoFinanciero(){}
+    public InstrumentoFinanciero(String nombre, double precio, Tipo tipo) {
+        this.nombre = nombre;
+        this.precio = precio;
+        this.tipo = tipo;
+    }
+    public double getPrecio() {
+        return precio;
+    }
+
+    public String getNombre() {
+        return nombre;
+    }
+
+    public Tipo getTipo() {
+        return tipo;
+    }
+
+    @Override
+    public String toString() {
+        return "InstrumentoFinanciero{" +
+                "nombre='" + nombre + '\'' +
+                ", precio=" + precio +
+                ", tipo=" + tipo +
+                '}';
+    }
+
+    public void setNombre(String nombre) {
+        if (nombre == null || nombre.isBlank()) {
+            throw new IllegalArgumentException("El nombre no puede ser nulo o vacío.");
+        }
+        this.nombre = nombre;
```

Comment:
Si el parámetro viene como null, este if va a dar una RuntimeException por nullpointer digamos al momento de preguntar por el isBlank(), no queda otra que encadenar o secuenciar las validaciones:

```suggestion
    public void setNombre(String nombre) {
        if (nombre == null ) {
            throw new IllegalArgumentException("El nombre no puede ser nulo .");
        } 
        if (! nombre.isBlank()){
            throw new IllegalArgumentException("El nombre no puede ser  vacío.")
        }
        this.nombre = nombre;
```